src/components/PageSettingsModal.jsx

```diff
@@ -0,0 +1,248 @@
+import React, { Component } from 'react';
+import axios from 'axios';
+import PropTypes from 'prop-types';
+import { Base64 } from 'js-base64';
+import * as _ from 'lodash';
+import elementStyles from '../styles/isomer-cms/Elements.module.scss';
+import {
+  frontMatterParser, concatFrontMatterMdBody, generatePageFileName,
+} from '../utils';
+
+// Constants
+const PERMALINK_REGEX = '^(/([a-z]+([-][a-z]+)*/)+)$';
+const permalinkRegexTest = RegExp(PERMALINK_REGEX);
+const PERMALINK_MIN_LENGTH = 4;
+const PERMALINK_MAX_LENGTH = 50;
+const TITLE_MIN_LENGTH = 4;
+const TITLE_MAX_LENGTH = 100;
+
+export default class PageSettingsModal extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      sha: '',
+      title: '',
+      permalink: '',
+      mdBody: '',
+      errors: {
+        title: '',
+        permalink: '',
+      },
+    };
+  }
+
+  async componentDidMount() {
+    try {
+      const { siteName, fileName, isNewPage } = this.props;
+
+      if (!isNewPage) {
+        const resp = await axios.get(`${process.env.REACT_APP_BACKEND_URL}/sites/${siteName}/pages/${fileName}`, {
+          withCredentials: true,
+        });
+        const { sha, content } = resp.data;
+
+        // split the markdown into front matter and content
+        const { frontMatter, mdBody } = frontMatterParser(Base64.decode(content));
+        const { title, permalink } = frontMatter;
+        this.setState({
+          sha, title, permalink, mdBody,
+        });
+      } else {
+        this.setState({ title: 'Title', permalink: '/permalink/' });
+      }
+    } catch (err) {
+      console.log(err);
+    }
+  }
+
+  saveHandler = async () => {
+    try {
+      const { siteName, fileName, isNewPage } = this.props;
+      const {
+        sha, title, permalink, mdBody,
+      } = this.state;
+
+      const frontMatter = { title, permalink };
+
+      // here, we need to re-add the front matter of the markdown file
+      const upload = concatFrontMatterMdBody(frontMatter, mdBody);
+
+      // encode to Base64 for github
+      const base64Content = Base64.encode(upload);
+      const newFileName = generatePageFileName(title);
+
+      if (isNewPage) {
+        // Create new page
+        const params = {
+          pageName: newFileName,
+          content: base64Content,
+        };
+
+        await axios.post(`${process.env.REACT_APP_BACKEND_URL}/sites/${siteName}/pages`, params, {
+          withCredentials: true,
+        });
+      } else if (newFileName === fileName) {
+        // A new file does not need to be created; the title has not changed
+        const params = {
+          content: base64Content,
+          sha,
+        };
+
+        // Update existing file
+        await axios.post(`${process.env.REACT_APP_BACKEND_URL}/sites/${siteName}/pages/${fileName}`, params, {
+          withCredentials: true,
+        });
+      } else {
+        // A new file needs to be created
+        // Delete existing page
```

Comment:
We can refactor this chunk of code by implementing an if condition on the deleting process, since this chunk of code is duplicated: 

```
        // Create new page
        const params = {
          pageName: newFileName,
          content: base64Content,
        };

        await axios.post(`${process.env.REACT_APP_BACKEND_URL}/sites/${siteName}/pages`, params, {
          withCredentials: true,
        });
      }
```

Reply:
cleaned up in 51f7fe6